lib/yt/models/resource.rb

```diff
@@ -13,7 +13,11 @@ class Resource < Base
       # @!attribute [r] id
       #   @return [String] the ID that YouTube uses to identify each resource.
       def id
-        @id ||= @match['id'] || fetch_channel_id
+        if @id.nil? && @match && @match[:kind] == :channel
```

Comment:
I would prefer to put this in `Channel`, but the way `URL` works, it instantiates a base `Resource` here, so fixing that is a much bigger refactor.